### DISCUSSION TITLE
Behaviour of relationship propagation to the inverse side

### DISCUSSION THREAD

**Discussion by rbao** [mikro-orm/mikro-orm#5750] [Q&A]
I was trying to upgrade mikro-orm in my project from 6.1.x to the latest 6.2.x. I have this two tests below, that use to pass but fails after the upgrade, and I narrowed down that between 6.1.9 and 6.1.10 something has changed that caused the test to break. The below two test I have will succeed for 6.1.9 but fails for any version after that without changing any of my project code (including 6.1.10).

In the below test, one JournalEntry has many LedgerEntry named as JournalEntry#lines. 

Journal Entry
```
  @OneToMany(() => LedgerEntry, (line) => line.journalEntry)
  lines = new Collection<LedgerEntry>(this);
```

Ledger Entry
```
  @ManyToOne(() => JournalEntry, { ref: true })
  journalEntry!: Ref<JournalEntry>;
```

The two tests:
```
    test('create with ManyToOne property', () => {
      const je = db.em.create(JournalEntry, {}, { partial: true });
      const line = db.em.create(LedgerEntry, { journalEntry: ref(je) }, { partial: true });

      expect(line.journalEntry.unwrap()).toBe(je);
      // in 6.1.9 je.lines.length return 1, but in 6.1.10+ je.lines.length return 0
      expect(je.lines.length).toBe(1);
    });

    test('assign to ManyToOne property should change the corresponding OneToMany field', () => {
      const je = db.em.create(JournalEntry, {}, { partial: true });
      const line = db.em.create(LedgerEntry, {}, { partial: true });
      line.journalEntry = ref(je);

      expect(line.journalEntry.unwrap()).toBe(je);
      // in 6.1.9 je.lines.length return 1, but in 6.1.10+ je.lines.length return 0
      expect(je.lines.length).toBe(1);
    });
```

If remember correctly in 5.x the relationship does not propagate to the inverse side, but since 6.x the relationship change started propagating to the inverse side by default, so I thought this is going to be the default behaviour, and during the upgraded to 6.1.x I added the above two test, but now it seems that in 6.1.10+ are going back to the 5.x behaviour where it no longer propagate to the inverse. My question is what is the intended long-term behaviour? Is mikro-orm suppose to do the inverse relationship propagation or not or I should always handle that on my own.

**Comment by rubiin** [comment]
this can be a breaking change . I am actually used to the 5.x behavior as thats what I read on the docs and havent visited that portion yet

**Comment by B4nan** [comment]
Propagation is not a new thing, it was implemented long before v5 actually.

Please provide a complete (and minimal) reproduction, without that it's hard to tell anything.

**Reply by rbao** [reply]
I am not reporting this as a bug, I am just asking a question what is the long-term intended behaviour. Is the intended behaviour to have inverse propagation or not? It's just a question, i.e is the intended behaviour for the above test to pass or not to pass? With that answer I can dive deeper to see if it's problem with my own code. Thank you

**Reply by B4nan** [reply]
Yes, propagation should work.

**Reply by rbao** [reply]
Looks like it is because I am using FlushMode.COMMIT and that is reason for the propagation to stop working.

I have a minimal repro here: https://github.com/rbao/mikro-orm-propagate which I cloned from mikro-orm/guide and added a failing test, you can check the last commit to see the test I added and run the test using `npm run test test/propagate.test.ts`.

If I use FlushMode.AUTO the test pass, if I use FlushMode.COMMIT the test will fail.

I believe that prior and including 6.1.9 changing the flush mode does not change the behaviour of inverse propagation, but after 6.1.9 changing flush mode will change the behaviour of inverse propagation. Is this an intended change?

**Reply by B4nan** [reply]
Hmm, you are correct, that perf optimization from v6.2.0 should have been applied only to scalars (just like the comment above it says).

This should fix it: a02c727233fab72a23f0fd4c4419f17ab552fd5d
